### PR TITLE
Add path template parsing

### PR DESCRIPTION
**Pull request by mbj**

### CODE REVIEW

**Comment by mbj** [comment]
@dkubb, @snusnu This is ready for review.

File: src/OpenApi/Paths.hs [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+import OpenApi.Prelude
+
+import qualified Data.Aeson      as JSON
+import qualified Data.Map.Strict as Map
+import qualified GHC.Enum        as GHC
+
+data Operation = Operation
+  { description :: OperationDescription
+  , operationId :: OperationId
+  , parameters  :: Maybe [Parameter]
```
Comment (dkubb) [inline src/OpenApi/Paths.hs:20]: I realize this is just a code move, but is it valid to have parameters equal to `[]`? Would it be handled differently from `Nothing`?

If it is impossible for it to be `[]` perhaps you could use the `Data.List.NonEmpty` type instead?

Not a blocker, just curious if a more constrained type could be used.

**Review by dkubb — APPROVED** [review]

File: src/OpenApi/Paths.hs (hunk 1 above)
Comment (mbj) [reply src/OpenApi/Paths.hs:20]: Yes, there are empty parameters.

Comment (mbj) [reply src/OpenApi/Paths.hs:20]: Think about "unique" (through their context) endpoints such as `GET /session`.

**Pull request merged**